src/test/java/server/RouterTest.java

```diff
@@ -1,52 +1,62 @@
 package test.java.server;
 
-import main.java.server.Router;
+import main.java.server.Response;
 import main.java.server.request.Request;
+import main.java.server.request.RequestParseException;
 import main.java.server.request.RequestParser;
+import main.java.server.router.ServerRouter;
 import org.junit.Before;
 import org.junit.Test;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.*;
 
 public class RouterTest {
 
-    private Router router;
+    private ServerRouter router;
     private RequestParser requestParser;
 
-    private static final String KNOWN_REQUEST = "GET /simple_get HTTP/1.1\r\n";
-    private static final String UNKNOWN_REQUEST = "GET /huh? HTTP/1.1\r\n";
-    private static final String OK = "HTTP/1.1 200 OK\r\n";
-    private static final String NOT_FOUND = "HTTP/1.1 404 Not Found\r\n";
+    private static final String KNOWN_GET_REQUEST = "GET /simple_get HTTP/1.1\r\n";
+    private static final String UNKNOWN_REQUEST = "OPTIONS /huh? HTTP/1.1\r\n";
+    private static final String HEAD_REQUEST = "HEAD /get_with_body HTTP/1.1\r\n";
+    private static final String OK_RESPONSE = "HTTP/1.1 200 OK\r\n";
+    private static final String NOT_FOUND_RESPONSE = "HTTP/1.1 404 Not Found\r\n";
 
     @Before
     public void setUp() {
-        router = new Router();
+        router = new ServerRouter();
         requestParser = new RequestParser();
     }
 
     @Test
-    public void returnsNotFoundIfPathIsUnknown() {
+    public void returnsNotFoundIfPathIsUnknown() throws RequestParseException {
         Request request = requestParser.create(UNKNOWN_REQUEST);
-        String response = router.route(request);
+        Response response = router.route(request);
 
-        assertEquals(NOT_FOUND, response);
+        assertEquals(NOT_FOUND_RESPONSE, response.getStatusLine());
     }
 
     @Test
-    public void returnsOKIfPathIsKnown() {
-        Request request = requestParser.create(KNOWN_REQUEST);
-        String response = router.route(request);
+    public void returnsOKIfPathIsKnown() throws RequestParseException {
+        Request request = requestParser.create(KNOWN_GET_REQUEST);
+        Response response = router.route(request);
 
-        assertEquals(OK, response);
+        assertEquals(OK_RESPONSE, response.getStatusLine());
     }
 
     @Test
-    public void doesNotReturnOkIfPathIsUnknown() {
+    public void doesNotReturnOkIfPathIsUnknown() throws RequestParseException {
         Request request = requestParser.create(UNKNOWN_REQUEST);
-        String response = router.route(request);
+        Response response = router.route(request);
 
-        assertNotEquals(OK, response);
+        assertNotEquals(OK_RESPONSE, response.getStatusLine());
```

Comment:
~Perhaps we could go further here. While it's useful to know that a function does not return undesirable output for given input, it usually ever more useful to know _what_ desirable output is returned for given input.~

Looks like we're already testing above that we get a `NOT_FOUND_RESPONSE` for the `UNKNOWN_REQUEST`, so maybe this test is unnecessary?